tests/stub/authorization.py

```diff
@@ -712,3 +714,114 @@ def get_vars(self, host=None):
+        !: AUTO RESET
+        !: ALLOW CONCURRENT
+
+        {+
+            {{
+                C: RUN "RETURN 1 as n" {} {"mode": "r"}
+                C: PULL {"n": 1000}
+                S: SUCCESS {"fields": ["n"]}
+                   FAILURE {"code": "Neo.ClientError.Security.AuthorizationExpired", "message": "Authorization expired"}
+                S: <EXIT>
+            ----
+                C: RUN "RETURN 2 as n" {} {"mode": "r"}
+                S: SUCCESS {"fields": ["n"]}
+                {+
+                    C: PULL {"n": 1}
+                    S: RECORD [1]
+                    SUCCESS {"type": "r", "has_more": true}
+                +}
+                C: DISCARD {"n": -1}
+                S: SUCCESS {}
+            ----
+                C: RUN "RETURN 3 as n" {} {"mode": "r"}
+                C: PULL {"n": "*"}
+                S: SUCCESS {"fields": ["n"]}
+                RECORD [1]
+                SUCCESS {"type": "r"}
+            }}
+
+        +}
+        """
+
+    @driver_feature(types.Feature.AUTHORIZATION_EXPIRED_TREATMENT)
+    def test_should_drop_connection_after_AuthorizationExpired(self):
+        self._server.start(
+            script=self.read_return_1_failure_return_2_and_3_succeed_script()
+        )
+        driver = Driver(self._backend, self._uri, self._auth,
+                        userAgent=self._userAgent)
+
+        session1 = driver.session('r', fetchSize=1)
+        session2 = driver.session('r')
+
+        session1.run('RETURN 2 as n').next()
+
+        try:
+            session2.run('RETURN 1 as n').next()
+        except types.DriverError as e:
+            self.assert_is_authorization_error(e)
+
+        session2.close()
+        session1.close()
+
+        accept_count = self._server.count_responses("<ACCEPT>")
+
+        # fetching another connection and run a query to force
+        # drivers which lazy close the connection do it
+        session3 = driver.session('r')
+        session3.run('RETURN 3 as n').next()
+        session3.close()
+
+        hangup_count = self._server.count_responses("<HANGUP>")
+
+        self.assertEqual(accept_count, hangup_count)
+        self.assertGreaterEqual(accept_count, 2)
+
+        driver.close()
+
+    @driver_feature(types.Feature.AUTHORIZATION_EXPIRED_TREATMENT)
+    def test_should_be_able_to_use_current_sessions_after_AuthorizationExpired(self):
+        self._server.start(
+            script=self.read_return_1_failure_return_2_and_3_succeed_script()
+        )
+
+        driver = Driver(self._backend, self._uri, self._auth,
+                        userAgent=self._userAgent)
+
+        session1 = driver.session('r', fetchSize=1)
+        session2 = driver.session('r')
+
+        session1.run('RETURN 2 as n').next()
+
+        try:
+            session2.run('RETURN 1 as n').next()
+        except types.DriverError as e:
+            self.assert_is_authorization_error(e)
+
+        session2.close()
+
+        session1.run('RETURN 2 as n').next()
```

Comment:
With the latest changes there is a little problem here. Java pulls all records for the first query before it runs the second one, but the script expects `DISCARD` in the end.

I made it work by introducing a fork:
```
{{
    C: DISCARD {"n": -1}
    S: SUCCESS {}
----
    C: PULL {"n": -1}
    S: RECORD [1]
       SUCCESS {}
}}
```

Or we could use an entirely separate script for this test.

Reply:
Thank you for adding this 👍